### ISSUE TITLE
Password generator seems to be broken with KeePassXC-Browser extension

### ISSUE THREAD

**Issue by clst** [smorks/keepassnatmsg#118]
The UI doesn't open any longer and when I click on the key icon. The password is filled as `[object Object]` There is nothing on the extension log that would provide any insight though.

Is it possible that this is the issue: https://github.com/keepassxreboot/keepassxc-browser/pull/1599

I am using keepassnatmsg 2.0.17 and The Firefox extension: KeePassXC-Browser 1.8.11 on Archlinux

All other functionality seems to be fine. I don't know when it last worked though. It started misbehaving some time last year.

EDIT: If I set KeePassXC Version override to 2 I get this instead:
![image](https://github.com/smorks/keepassnatmsg/assets/293237/374dab25-3082-438b-a83b-ece27ef1c910)

**Comment by clst** [comment]
OK. Sorry for not investigating more. The correct version for the password generator protocol in Keepass 2.54 seems to be ~2.6

It works great with `2.6` but the Browser Extension complains about an old KeePassXC.

Is that a change that needs to be done in the plugin, or is XC now just incompatible with the original KeePass 2.x for password generation?

**Comment by smorks** [comment]
> Is that a change that needs to be done in the plugin, or is XC now just incompatible with the original KeePass 2.x for password generation?

yeah, it looks like a change would need to have to be done for password generation to work properly. if I ever have some free time I can look into it, but it's not super high on my priority list.

relavant protocol info: https://github.com/keepassxreboot/keepassxc-browser/blob/develop/keepassxc-protocol.md#generate-password
